src/app/reporting/Engine.cpp

```diff
@@ -336,6 +367,66 @@ void Engine::Run()
         mCurReadHandlerIdx = (mCurReadHandlerIdx + 1) % CHIP_IM_MAX_NUM_READ_HANDLER;
         readHandler        = imEngine->mReadHandlers + mCurReadHandlerIdx;
     }
+
+    bool allReadClean = true;
+    for (auto & handler : InteractionModelEngine::GetInstance()->mReadHandlers)
```

Comment:
I think there is a case here, where on every iteration through `Run()`, you're might end up in a situation where you'll keep generating the same report to a particular subscriber, and never ever progress to generating reports for other subscribers.

The situation happens I think, when:
* You were not able to finish dispatching all reports to all subscribers in a single execution of `Run()`. Consequently, the global change set is unchanged upon completion of `Run()`.
* You were not able to finish dispatching reports because you were not able to fit all attribute data into a single message. `BuildSingleReportDataAttributeDataList` errors out, but leaves its dirty bit enabled on the ReadHandler.

On the next run, it's going to go through the same motions, error'ing out building a report to the same handler.

Don't think this is a use-case that will be encountered immediately, but it's worth thinking filing it as a follow-up issue/PR.

Reply:
Yep, file the ticket here, https://github.com/project-chip/connectedhomeip/issues/9553, thanks